### PR TITLE
fix(dms/rabbitmq): fix the problem that single instance query is empty

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The RabbitMQ instance for single type has no 'product_infos' structure.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. fix the problem that single instance query is empty.
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
NONE
```

### CODE REVIEW

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 1]
```diff
@@ -312,7 +312,7 @@ func getProductDetail(config *config.Config, d *schema.ResourceData, engineType
 		}
 		for _, v := range ps.Values {
 			for _, p := range v.Details {
-				if engineType == "kafka" {
+				if v.Name == "single" || engineType == "kafka" {
```
Comment (niuzhenguo) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: This seems a bit difficult for understanding, how about just checking rabbitmq engineType and not single for response with ProductInfos like below?
```
if engineType == "rabbitmq" && v.Name == "cluster"
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsRabbitmqInstances_single'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsRabbitmqInstances_single -timeout 360m -parallel 4
=== RUN   TestAccDmsRabbitmqInstances_single
=== PAUSE TestAccDmsRabbitmqInstances_single
=== CONT  TestAccDmsRabbitmqInstances_single
--- PASS: TestAccDmsRabbitmqInstances_single (657.95s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms 658.040s
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: I added a note for this code

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 2]
```diff
@@ -312,7 +312,9 @@ func getProductDetail(config *config.Config, d *schema.ResourceData, engineType
 		}
 		for _, v := range ps.Values {
 			for _, p := range v.Details {
-				if engineType == "kafka" {
+				// All informations of product for single instance type and the kafka engine type are stored in the
```
Comment (hmbig2) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: the new API is:  https://dms.cn-north-4.myhuaweicloud.com/v2/kafka/products
the struct is the same for single and cluster

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: Although this works, I think we should write it like this
```
if engineType == "kafka" {
    ...
} else if engineType == "rabbitmq" {
    if v.Name == "single" {
        ...
    } else {
        ...
    }
}
```
to make the logic clear

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccDmsRabbitmqInstances_single'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccDmsRabbitmqInstances_single -timeout 360m -parallel 4
=== RUN   TestAccDmsRabbitmqInstances_single
=== PAUSE TestAccDmsRabbitmqInstances_single
=== CONT  TestAccDmsRabbitmqInstances_single
--- PASS: TestAccDmsRabbitmqInstances_single (655.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       655.946s
```
```
make testacc TEST='./huaweicloud/services/acceptance/dms' TESTARGS='-run=TestAccKaf
kaInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms -v -run=TestAccKafkaInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_basic
--- PASS: TestAccKafkaInstance_basic (1279.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms    1279.147s
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: The function has been split into two functions

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: The function has been split into two functions

**Pull request merged**